processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -782,6 +782,13 @@ else if ( candidate.getAccessorType() == AccessorType.GETTER ) {
                     // an adder has been found (according strategy) so overrule current choice.
                     candidate = adderMethod;
                 }
+
+                if ( cmStrategy == CollectionMappingStrategyGem.TARGET_IMMUTABLE
```

Comment:
Hmmm. The [CollectionMappingStrategy](https://github.com/mapstruct/mapstruct/blob/main/core/src/main/java/org/mapstruct/CollectionMappingStrategy.java) documentation is pretty lame reading it at hindsight.  I've been reading it a couple of times, but its intended usage (e.g. JPA?) does not really become clear from the text.. does it? And what is the default behavior (if you omit the strategy all together)? 

Also, I guess the description for each value should describe the "regular" and the "update" behavior separately.

Reply:
Why limit this only to  the `TARGET_IMMUTABLE` option? I would say that in general the presence of only a `getter` does not constitute a reason to give a "unmapped target property" message.  In my opinion, only when an accessor with the clear intention of writing (`setter`, `adder`) is there with no corresponding source (mapping) the "unmapped target property" message should be given.

I guess that the behavior is even backward compatible. For existing usage the users have "fixed" such errors. We simply drop the error in new usage (clear release notes though).

Reply:
We have an extensive documentation about the `CollectionMappingStrategy` (https://mapstruct.org/documentation/stable/reference/html/#collection-mapping-strategies). By default it is accessor only (check `Mapper#collectionMappingStrategy`. I think that the table in the docs is quite clear, perhaps we need to add that table to the javadoc.

> I would say that in general the presence of only a `getter` does not constitute a reason to give a "unmapped target property" message. 

I would tend to agree with you on this one. However, if we remove this here it would mean that there won't be any target property that you can map. And for collections and map we do support things like `target.getNames().add(name)` and also for update mappings doing things like
```
target.getNames().clear()
target.getNames().addAll()
```

This would be a rather larger breaking change, since getters will no longer be considered for populating collections.

Reply:
> perhaps we need to add that table to the javadoc.

That would be a good idea 👍

Reply:
> This would be a rather larger breaking change, since getters will no longer be considered for populating collections.

But I guess there are two separate things:
1.  using a getter when present: no problem with that.. keep the existing functionality
2. generating  a  "unmapped target property" fault when a getter is there and there's no corresponding source property. That's the thing I like to address in a more generic way.

Reply:
>That would be a good idea 👍

I'll add the table.

>generating a "unmapped target property" fault when a getter is there and there's no corresponding source property. That's the thing I like to address in a more generic way.

If we do this, isn't this a bit of a strange behaviour for the user? They had no warning / error before. They add a getter on the source side and not all of a sudden it gets mapped. It would be confusing in my opinion.

Reply:
or for this specific case we change from an error to a warning..

Reply:
@sjaakd I've added the table to the javadoc. 

Regarding the other stuff, I think that it makes more sense to do it in a different issue.  Can you create it or should we wait for someone to actually need this?

Reply:
I'm afraid we'll forget that.. To be honest. My comments dated from a month ago and I forgot already.. Anyway.. I leave it to you if you want to fix that.